3684/1/yaml_reader.rb

```diff
@@ -0,0 +1,52 @@
+require 'yaml'
+require 'gems'
+
+class YamlReader
+  attr_reader :file
+
+  def initialize(filepath)
+    @filepath = filepath
+  end
+
+  def parse
+    adress_of_source_code(read_yaml)
+  end
+
+  private
+
+  def read_yaml
+    file = YAML.load_file("#{@filepath}.yaml")
+    file['gems'].to_s.split(' ').each do |item|
+      item.delete!('-')
+    end
+  rescue IncorrectFileError
+    puts 'File doesn\'t exist'
+    abort
+  end
+
+  def correct(adresses)
+    adresses.map do |adress|
+      make_redirection(adress)
+    end
+  end
+
+  def make_redirection(adress)
+    if adress[0..4].include?('s')
+      adress
+    else
+      string = adress[4..adress.length]
+      "https#{string}"
+    end
+  end
+
+  def adress_of_source_code(gems)
+    adresses = []
+    gems.each do |item|
+      adresses << Gems.info(item)['source_code_uri']
+    end
+    correct(adresses)
+  end
+end
+
+class IncorrectFileError < StandardError
```

Comment:
1 empty classes defines in one line `class A; end`
2 as soon as you fix error class above :up you gonna remove this class definition